test-interface/src/main/scala/org/scalajs/testinterface/TestDetector.scala

```diff
@@ -53,7 +53,7 @@ private[scalajs] object TestDetector {
     def tryLoad(name: String): Option[Framework] =
       tryLoadFromReflect(name).orElse(tryLoadFromExportsNamespace(name))
 
-    names.toStream.map(tryLoad).flatten.headOption
+    names.iterator.map(tryLoad).find(_.isDefined).flatten
```

Comment:
have you considered using `collectFirst { Some(f) => f }` for better readability?